framework/python/src/common/session.py

```diff
@@ -335,6 +364,27 @@ def add_test_result(self, result):
       result.result = 'In Progress'
       self._results.append(result)
 
+  def set_test_result_error(self, result):
+    """Set test result error"""
+    updated = False
+
+    for test_result in self._results:
+
+      # result type is TestCase object
+      if test_result.name == result.name:
+
+        # Just update the result and description
+        if test_result.result == 'In Progress':
+          test_result.result = 'Error'
+          test_result.description = result.description
+          test_result.recommendations = None
+          updated = True
+
+    if not updated:

```

Comment:
Is there a reason why it would not be updated in the check above? Why not just use this part standalone?

Reply:
fixed